pyccel/parser/syntactic.py

```diff
@@ -1119,6 +1119,10 @@ def _visit_comprehension(self, stmt):
 
         self.exit_loop_scope()
 
+        if stmt.ifs:
+            errors.report("Can't calculate list length when comprehension contains if statements.\n" + PYCCEL_RESTRICTION_TODO,
```

Comment:
I found this error a bit confusing, the error assumes that i know the issue.
(i did understand it after I read the issue)

Reply:
Hmm, I was hoping that the line number would help make it obvious. Do you have another suggestion?

Reply:
```suggestion
            errors.report("Cannot handle if statements in list comprehensions. List length cannot be calculated.\n" + PYCCEL_RESTRICTION_TODO,
```
?

Reply:
I think This describe the Problem clearly. `Cannot handle if statements in list comprehensions.`
which for me was hard to know from the start with the previous message.

Reply:
Fixed